registry/storage/garbagecollect.go

```diff
@@ -169,6 +169,10 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 
 	for repo, dgsts := range deleteLayerSet {
 		for _, dgst := range dgsts {
+			emit("%s: layer link eligible for deletion: %s", repo, dgst)
```

Comment:
Is there a test we could add that would catch this in the future so we don't have someone encountering this when testing an actual release?

Reply:
I think it's rather "complicated" because the GC code has been largely unmaintained. At some point, I suggested we move it to an `internal` package until there is some plan to address a lot of the issues the users report. Fixing it up would be a serious effort.

Reply:
Will the GC be refactored or redesigned? Currently, it only supports **offline** GC, which I think is unacceptable in a production environment. Additionally, the lookup performance is relatively low, and for the S3 driver, it will make a large number of S3 requests.

Reply:
Not sure. It's a nontrivial amount of work to make things work online, especially in high volume traffic / large storage environments the current design doesn't scale.

Reply:
Sure, online GC needs to be redesigned, and it requires a significant amount of work. Is there a plan to start with some research? If there is any work in this area in the future, please include me in the relevant loop. Thanks in advance.

Reply:
There is no plan at the moment I'm afraid. This project suffers from zero funding and thus, a general lack of interest on the current maintainers' side bar few exceptions. I am not sure if anything beyond the `v3` release will happen in this project.